config/externalname.go

```diff
@@ -2853,6 +2853,11 @@ func ExternalNameConfigurations() config.ResourceOption {
 		if e, ok := ExternalNameConfigs[r.Name]; ok {
 			r.Version = common.VersionV1Beta1
 			r.ExternalName = e
+			// Note(turkenh): This is special to provider-aws. We had injected
+			// region as a parameter for all resources to be consistent with
+			// the native aws provider, and now, we need to add manually it to
+			// the identifier fields for all resources.
+			r.ExternalName.IdentifierFields = append(r.ExternalName.IdentifierFields, "region")
```

Comment:
My understanding is that we do not need to treat the `iam` group specially because although `spec.forProvider.region` is declared as an identifier field because that field is missing for the managed resources in that group, it becomes ineffective.